presto-docs/src/main/sphinx/functions/aggregate.rst

```diff
@@ -355,3 +360,75 @@ Lambda Aggregate Functions
         -- (2, 42)
 
     The state type must be a boolean, integer, floating-point, or date/time/interval.
+
+
+Filtering During Aggregation
+----------------------------
+
+The ``FILTER`` keyword can be used to remove rows from aggregation processing
+with a condition expressed using a ``WHERE`` clause. This is evaluated for each
+row before it is used in the aggregation and is supported for all aggregate
+functions.
+
+    AGGREATE_FUNCTION(...) FILTER (WHERE <condition>)
+
+A common and very useful example is to use ``FILTER`` to remove nulls from
+consideration when using ``array_agg``.
+
+.. code-block:: none
+
+    SELECT array_agg(col) FILTER (WHERE col IS NOT NULL)
+    FROM t
+
+As another example, imagine you want to add a condition on the count for Iris
+flowers, modifying the following query.
+
+.. code-block:: none
+
+    SELECT species,
+           count(*) AS count
+    FROM memory.default.iris
+    GROUP BY species;
+
+.. code-block:: none
+
+    species    | count
+    -----------+-------
+    setosa     |   50
+    virginica      50
+    versicolor|    50
+
+If you just use a normal WHERE statement you loose information:
+
+.. code-block:: none
+
+    SELECT species,
+        count(*) AS count
+    FROM memory.default.iris
+    WHERE petallengthcm > 4
```

Comment:
`petal_length_cm`

Reply:
in the book we have column names that conflate it all into one word for all of them ... if we change this here we should change all queries in the book. Wdyt @mattsfuller  and @martint ? For reference https://github.com/starburstdata/presto-the-definitive-guide/tree/master/iris-data-set

I am fine changing this but this has to be done next week in the book source before it finished copy edit so we need to decide ASAP.

Reply:
I think it's much cleaner and readable with what @electrum suggested, so I suggest we change it.

Reply:
okay ... I will update this and the book repo now. and the book itself next week